src/coreComponents/linearAlgebra/interfaces/hypre/HypreSuiteSparse.cpp

```diff
@@ -229,68 +234,80 @@ int SuiteSparseSolve( SuiteSparseData & SSData,
+      if( status < 0 )
+      {
+        umfpack_dl_report_info( SSData.Control, SSData.Info );
+        umfpack_dl_report_status( SSData.Control, status );
+        GEOSX_ERROR( "Hypre SuiteSparse interface: umfpack_dl_solve failed." );
+      }
+      sol_Vector = ( HYPRE_Vector ) sol_vector;
     }
-    sol_Vector = ( HYPRE_Vector ) sol_vector;
   }
 
   // Copy partitioning otherwise hypre would point to data in x (issues with deallocation!!!)
   HYPRE_BigInt * partitioning = hypre_CTAlloc( HYPRE_BigInt, 2, HYPRE_MEMORY_HOST );
   partitioning[0] = hypre_ParVectorPartitioning( x.unwrapped() )[0];
   partitioning[1] = hypre_ParVectorPartitioning( x.unwrapped() )[1];
 
-  HYPRE_ParVector sol_ParVector;
-  GEOSX_LAI_CHECK_ERROR( HYPRE_VectorToParVector( x.getComm(),
-                                                  sol_Vector,
-                                                  partitioning,
-                                                  &sol_ParVector ) );
-  std::copy( hypre_VectorData( hypre_ParVectorLocalVector( sol_ParVector ) ),
-             hypre_VectorData( hypre_ParVectorLocalVector( sol_ParVector ) ) + x.localSize(),
-             x.extractLocalVector() );
+  if( SSData.comm != MPI_COMM_NULL )
+  {
+    HYPRE_ParVector sol_ParVector;
+    GEOSX_LAI_CHECK_ERROR( HYPRE_VectorToParVector( SSData.comm,
+                                                    sol_Vector,
+                                                    partitioning,
+                                                    &sol_ParVector ) );
+    std::copy( hypre_VectorData( hypre_ParVectorLocalVector( sol_ParVector ) ),
+               hypre_VectorData( hypre_ParVectorLocalVector( sol_ParVector ) ) + x.localSize(),
+               x.extractLocalVector() );
+    GEOSX_LAI_CHECK_ERROR( hypre_ParVectorDestroy( sol_ParVector ) );
+  }
+
   GEOSX_LAI_CHECK_ERROR( HYPRE_VectorDestroy( sol_Vector ) );
-  GEOSX_LAI_CHECK_ERROR( hypre_ParVectorDestroy( sol_ParVector ) );
   GEOSX_LAI_CHECK_ERROR( hypre_SeqVectorDestroy( b_vector ) );
 
   time = watch.elapsedTime();
-  MpiWrapper::bcast( &time, 1, SSData.workingRank, SSData.comm );
+  MpiWrapper::bcast( &time, 1, SSData.workingRank, SSData.globalComm );
 
   return status;
 }
 
 void SuiteSparseDestroy( SuiteSparseData & SSData )
 {
-  int const rank = MpiWrapper::Comm_rank( SSData.comm );
-  if( rank == SSData.workingRank )
+  if( SSData.comm != MPI_COMM_NULL )
   {
-    umfpack_dl_free_symbolic( &SSData.Symbolic );
-    umfpack_dl_free_numeric( &SSData.Numeric );
-    delete [] SSData.rowPtr;
-    delete [] SSData.colIndices;
-    delete [] SSData.data;
-    GEOSX_LAI_CHECK_ERROR( hypre_CSRMatrixDestroy( SSData.CSRmatrix ) );
+    int const rank = MpiWrapper::Comm_rank( SSData.comm );
+    if( rank == SSData.workingRank )
+    {
+      umfpack_dl_free_symbolic( &SSData.Symbolic );
+      umfpack_dl_free_numeric( &SSData.Numeric );
+      delete [] SSData.rowPtr;
+      delete [] SSData.colIndices;
+      delete [] SSData.data;
```

Comment:
We should make these `vector` or `array1d`

Reply:
Agreed. Let's tackle everything in a separate PR.